File: backend/src/utils/currency.py
"""
Currency formatting utility for backend
Single source of truth for PKR currency display
Format: "Rs. X,XXX" with thousands separator, decimals stripped if .00
"""

def format_price(amount: float) -> str:
    """
    Format amount as PKR currency string.

    Args:
        amount: Numeric amount to format

    Returns:
        Formatted string like "Rs. 4,999" or "Rs. 24.99"
    """
    # Handle edge cases
    if amount is None or not isinstance(amount, (int, float)):
        return "Rs. 0"

    # Round to 2 decimals
    rounded = round(amount, 2)

    # Check if it's a whole number (or effectively .00)
    is_whole_number = rounded == int(rounded)

    if is_whole_number:
        # Format without decimals: 4999 → "Rs. 4,999"
        integer_part = int(rounded)
        formatted = f"{integer_part:,}"
        return f"Rs. {formatted}"
    else:
        # Format with 2 decimals: 24.99 → "Rs. 24.99"
        formatted = f"{rounded:,.2f}"
        return f"Rs. {formatted}"

File: frontend/artifacts/zaitoun-loralai/src/utils/currency.ts
/**
 * Currency formatting utility
 * Single source of truth for PKR currency display across the app
 * Format: "Rs. X,XXX" with thousands separator, decimals stripped if .00
 */

export function formatPrice(amount: number): string {
  // Handle edge cases
  if (!Number.isFinite(amount)) return 'Rs. 0';

  // Round to 2 decimals
  const rounded = Math.round(amount * 100) / 100;

  // Check if it's a whole number (or effectively .00)
  const isWholeNumber = rounded % 1 === 0;

  if (isWholeNumber) {
    // Format without decimals: 4999 → "Rs. 4,999"
    const formatted = Math.floor(rounded).toLocaleString('en-US');
    return `Rs. ${formatted}`;
  } else {
    // Format with 2 decimals: 24.99 → "Rs. 24.99"
    const formatted = rounded.toLocaleString('en-US', {
      minimumFractionDigits: 2,
      maximumFractionDigits: 2,
    });
    return `Rs. ${formatted}`;
  }
}
